Makefile

```diff
@@ -52,65 +52,26 @@ printvars: ## Print the Environment variables present in calls to make, plus var
 	@printf "\n==== Environment Variables ====\n\n"
 	@printenv
 
-.python-version: ## Attempt to setup python using pyenv
-	@if ! command -v pyenv &> /dev/null; then \
-		echo "WARNING: pyenv could not be found. Install pyenv to get a virtual env running with the compatible python version: ${PYTHON_VERSION}. Will fallback to using system python3."; \
-	else \
-	  	set -x; \
-	  	echo "pyenv setting python version to ${PYTHON_VERSION}"; \
-  		pyenv install -s ${PYTHON_VERSION}; \
-  		pyenv local ${PYTHON_VERSION}; \
-  		python3 -V; \
-  		if [ "$$(python3 -V)" != "Python ${PYTHON_VERSION}" ]; then \
-  			echo "ERROR: pyenv was not able to set local python version to ${PYTHON_VERSION}"; \
-  			exit 1; \
-  		fi; \
-	fi;
-
-.venv: ## Ensure a virtual environment is established at .venv
-	@( \
-		set -x; \
-		test -d .venv || python3 -m venv .venv/${venv_name}; \
-	)
-
-.PHONY: requirements-dev
-upgrade :=  #unset it
-requirements-dev: .venv ## Install pip packages in dev virtual env. Add upgrade=true to upgrade required packages to newest version (can be lengthy)
-	# Because this depends on .venv, the virtual env should exist to activate
-	# This will "activate" the virtual env only for the duration of the scripts in the parens-scope
-	# Then when this make rule recipe is complete, the virtual env will be dropped
-	# But it IS there and populated
-	# Must _manually_ reactivate the virtual env to interact with it on the command line
-	@( \
-		source .venv/${venv_name}/bin/activate; \
-		echo "virtual env at .venv/${venv_name} activated (temporarily)"; \
-		pip install $$(cat requirements/requirements-dev.txt | grep 'pip=='); \
-		src_roots=(${src_root_paths}); \
-		for src_root in "$${src_roots[@]}"; do \
-			pip install ${if ${upgrade},--upgrade,} --editable "$${src_root}[dev]"; \
-		done; \
-	)
+uv-sync:
+	uv sync --no-cache --extra dev --locked
 
 .ivy2: ## Ensure user has a ~/.ivy2 dir, which will be bound to in a docker container volume to save on dependency downloads
 	@mkdir -p ~/.ivy2
 
-.PHONY: activate
-activate: ## Spit out the command to run to activate the virtual env, since you can't do it within a make shell process. Use this like: source $(make activate)
-	@echo ".venv/${venv_name}/bin/activate"
-
 .PHONY: local-dev-setup
-local-dev-setup: .python-version requirements-dev check-dependencies .ivy2 ## Setup python, virtual environment, and pip dependencies, then check version info
+local-dev-setup: uv-sync check-dependencies .ivy2 ## Setup virtual environment and pip dependencies, then check version info
 
 .PHONY: check-dependencies
 check-dependencies: ## Prints out the versions of dependencies in use
```

Comment:
If I am following correctly, uv will be managing the virtual environment and remove the need for us doing it ourself?